pkg/controller/core.oam.dev/v1alpha2/application/generator.go

```diff
@@ -95,6 +97,9 @@ func (h *AppHandler) GenerateApplicationSteps(ctx monitorContext.Context,
 			Apply:  h.Dispatch,
 			Delete: h.Delete,
 		})
+	configprovider.Install(handlerProviders, h.r.Client, func(ctx context.Context, resources []*unstructured.Unstructured, applyOptions []apply.ApplyOption) error {
```

Comment:
Move to after `generator.GenerateRunners`, and make sure this provider can not be replaced.

Reply:
No, it must register first. The registered job won't be replaced when exists.